tests/mocha/toolbox_test.js

```diff
@@ -0,0 +1,245 @@
+      toolbox.horizontalLayout_ = true;
+      toolbox.position();
+      checkHorizontal(toolbox);
+      chai.assert.equal(toolbox.HtmlDiv.style.bottom, '0px', 'Check bottom');
+    });
+    test('Position Vertical at right', function() {
+      var toolbox = this.toolbox;
+      toolbox.toolboxPosition = Blockly.TOOLBOX_AT_RIGHT;
+      toolbox.horizontalLayout_ = false;
+      toolbox.position();
+      chai.assert.equal(toolbox.HtmlDiv.style.right, '0px', 'Check right');
+      checkVertical(toolbox);
+    });
+    test('Position Vertical at left ', function() {
+      var toolbox = this.toolbox;
+      toolbox.toolboxPosition = Blockly.TOOLBOX_AT_RIGHT;
+      toolbox.horizontalLayout_ = false;
+      toolbox.position();
+      chai.assert.equal(toolbox.HtmlDiv.style.left, '0px', 'Check left');
+      checkVertical(toolbox);
+    });
+  });
+
+  suite('syncTrees_', function() {
+    setup(function() {
+      this.tree = new Blockly.tree.TreeControl(this.toolbox, this.toolbox.config_);
+      this.tree.blocks = [];
+      this.toolboxXml = document.getElementById('toolbox-test');
+      this.separatorIdx = 0;
+      this.buttonIdx = 1;
+      this.dynamicCategoryIdx = 3;
+      this.categorySeparatorIdx = 2;
+    });
+    test('Having a dynamic category', function() {
+      this.toolbox.syncTrees_(this.toolboxXml, this.tree,
+          this.toolbox.workspace_.options.pathToMedia);
+      chai.assert.equal(this.tree.children_[this.dynamicCategoryIdx].blocks, 'VARIABLE');
+    });
+    test('Node is expanded', function() {
+      var openNode = this.toolbox.syncTrees_(this.toolboxXml, this.tree,
+          this.toolbox.workspace_.options.pathToMedia);
+      chai.assert.exists(openNode);
+    });
+    test('Having a tree separator', function() {
+      this.toolbox.syncTrees_(this.toolboxXml, this.tree,
+          this.toolbox.workspace_.options.pathToMedia);
+      var sepString = Blockly.utils.xml.domToText(this.tree.children_[0].blocks[this.separatorIdx]);
+      chai.assert.equal(sepString, '<sep xmlns="http://www.w3.org/1999/xhtml" gap="-1"></sep>');
+    });
+    test('Separator between two categories', function() {
+      this.toolbox.syncTrees_(this.toolboxXml, this.tree,
+          this.toolbox.workspace_.options.pathToMedia);
+      chai.assert.isTrue(this.tree.children_[
+          this.categorySeparatorIdx] instanceof Blockly.Toolbox.TreeSeparator);
+    });
+    test('Having a button', function() {
+      this.toolbox.syncTrees_(this.toolboxXml, this.tree,
+          this.toolbox.workspace_.options.pathToMedia);
+      var btnString = Blockly.utils.xml.domToText(this.tree.children_[0].blocks[this.buttonIdx]);
+      chai.assert.equal(btnString,
+          '<button xmlns="http://www.w3.org/1999/xhtml" text="insert" callbackkey="insertConnectionRows"></button>');
+    });
+    test('Colours are set using style', function() {
+      sinon.stub(this.toolbox, "setColourFromStyle_");
+      sinon.stub(this.toolbox, "setColour_");
```

Comment:
I'm not sure if this is the case for any of the other tests, but I'd be a bit hesitant on using stubs for the class you're testing. In this tests, if there were any bugs in `setColour_` and `setColourFromStyle_` then they wouldn't be caught, so its a bit misleading. Unless testing the logic for setColour is very complex, I think you should test it, otherwise maybe rename the test to say something around the lines of that it's checking that  the internal set colour methods are called by `syncTrees`.

If you wanted to test the logic AND how many times the method was called, you could use `sinon.spy`

Reply:
There are quite a few more tests that would need to be written to properly test setColour and setStyle. I wasn't trying to go that in depth at this point in time. I updated the name so hopefully it isn't misleading for what i am trying to test.